packages/ai/src/ui/chat.ts

```diff
@@ -414,7 +415,17 @@ export abstract class AbstractChat<UI_MESSAGE extends UIMessage> {
    * Attempt to resume an ongoing streaming response.
    */
   resumeStream = async (options: ChatRequestOptions = {}): Promise<void> => {
-    await this.makeRequest({ trigger: 'resume-stream', ...options });
+    if (this.isResuming) {
+      return;
+    }
+
+    this.isResuming = true;
+
+    try {
+      await this.makeRequest({ trigger: 'resume-stream', ...options });
+    } finally {
+      this.isResuming = false;
+    }
```

Comment:
I think the implementation might lead to unexpected behavior. When called the 2nd time, it will resolve immediately.

Instead of implementing a `true/false` flag, I would prefer we implement `this.resumeRequestPromise` that is set to `null` by default and after the resume request concludes. If it's truthy, return it. That way a 2nd parallel call will also await the resume request